File: parsers/template.py
from handlers.soup import getSoupObject

def getCodeTemplate(filepath: str) -> None:
    soup = getSoupObject(filepath)
    data = soup.find("div", class_="view-lines monaco-mouse-cursor-text").find_all("div")

    template = ""
    for line in data:
        line = line.find("span")
        for word in line:
            template += word.text

        template += "\n"

    return template

File: _parsers/template.js
export async function parseCodeTemplate(html) {
    return new Promise((resolve, reject) => {
        try {
            const parser = new DOMParser();
            const doc = parser.parseFromString(html, "text/html");

            let template = "";
            const codeLines = doc.querySelectorAll(".view-lines.monaco-mouse-cursor-text div");

            codeLines.forEach((div) => {
                template += div.querySelectorAll("span")[0].textContent + "\n";
            });
            
            resolve(template);
        } catch (error) {
            console.error("Error fetching or parsing page:", error);
            reject(error);
            return false;
        }
    });
}

